scapy/contrib/isotp.py

```diff
@@ -1780,17 +1928,19 @@ def __init__(self,
                      padding=False,
                      transmit_time=100,
                      basecls=ISOTP):
+            # type: (Optional[Union[str, SuperSocket]], int, int, Optional[int], Optional[int], bool, bool, int, Type[Packet]) -> None  # noqa: E501
 
             if not isinstance(iface, six.string_types):
+                iface = cast(SuperSocket, iface)
```

Comment:
Why would you call a socket `iface` ?!

Reply:
This is for interoperability with ISOTPSoftSockets.  If a NativeCANSocket is provided, the interface name of this socket is extracted and an ISOTPNativeSocket will be opened on this interface.